tfhe/src/high_level_api/array/dyn_boolean_backend.rs

```diff
@@ -0,0 +1,267 @@
+//! This module contains the implementation of the FheBool array backend
+//! where the location of the values and computations can be changed/selected at runtime
+use super::cpu_boolean_backend::{CpuFheBoolArrayBackend, FheBoolId};
+use super::helpers::{create_sub_mut_slice_with_bound, create_sub_slice_with_bound};
+use super::traits::{
+    ArrayBackend, BackendDataContainer, BackendDataContainerMut, BitwiseArrayBackend,
+    ClearBitwiseArrayBackend,
+};
+use super::{FheBackendArray, FheBackendArraySlice, FheBackendArraySliceMut};
+
+use crate::integer::BooleanBlock;
+use crate::prelude::{FheDecrypt, FheTryEncrypt};
+use crate::{ClientKey, Device};
+use std::borrow::{Borrow, Cow};
+use std::ops::RangeBounds;
+
+pub type FheBoolArray = FheBackendArray<DynFheBoolArrayBackend, FheBoolId>;
+pub type FheBoolSlice<'a> = FheBackendArraySlice<'a, DynFheBoolArrayBackend, FheBoolId>;
+pub type FheBoolSliceMut<'a> = FheBackendArraySliceMut<'a, DynFheBoolArrayBackend, FheBoolId>;
+
+pub struct DynFheBoolArrayBackend;
+
+impl ArrayBackend for DynFheBoolArrayBackend {
```

Comment:
the dyn backend is the one that could mix devices right ?

Reply:
The one where you can select and change device at runtime